src/daft-sql/src/planner.rs

```diff
@@ -165,7 +194,7 @@ impl SQLPlanner {
             }
 
             let name = ident_to_str(&cte.alias.name);
-            let plan = self.plan_query(&cte.query)?;
+            let plan = self.new_with_context().plan_query(&cte.query)?;
```

Comment:
why do we need to create a new context here?

Reply:
I don't think the alias or table maps are shared with the CTE which is why I created a new planner. `new_with_context` shares the context (catalog + ctes) but nothing else.